lib/strategy/dns_poll.ex

```diff
@@ -151,6 +151,11 @@ defmodule Cluster.Strategy.DNSPoll do
     []
   end
 
+  def lookup_all_ips(q) do
+    Enum.map([:a, :aaaa], fn t -> :inet_res.lookup(q, :in, t) end) |> List.flatten
+  end
+
   # turn an ip into a node name atom, assuming that all other node names looks similar to our own name
-  defp format_node({a, b, c, d}, base_name), do: :"#{base_name}@#{a}.#{b}.#{c}.#{d}"
+  defp format_node(ip, base_name), do: :"#{base_name}@#{ip |> :inet_parse.ntoa |> to_string()}"
+
```

Comment:
The to string conversion is automatic on interpolation:

```suggestion
  defp format_node(ip, base_name), do: :"#{base_name}@#{:inet_parse.ntoa(ip)}"
```

:)

Reply:
That is nicer, but pipes are so fun.